### ISSUE TITLE
[bug] Multiple `--build=missing:~foo` does not work

### ISSUE THREAD

**Issue by rconde01** [conan-io/conan#16323]
### Describe the bug

Conan 2.3.1
I wanted to build all missing packages except for specified packages. I tried:
```
conan graph build-order --reduce --order-by recipe --build=missing:~foo1/*
```
This worked. However the following cases did not work:
```
conan graph build-order --reduce --order-by recipe --build=missing:~foo1/* --build=missing:~foo2/* 
conan graph build-order --reduce --order-by recipe --build=missing --build=missing:~foo1/*
conan graph build-order --reduce --order-by recipe --build=missing:* --build=missing:~foo1/*
conan graph build-order --reduce --order-by recipe --build=missing --build=~foo1/*
```

[conan_build_missing_multiple_negate.zip](https://github.com/conan-io/conan/files/15407836/conan_build_missing_multiple_negate.zip)

### How to reproduce it

Run `test.py` for the attached test

**Comment by AbrilRBS** [comment]
As I was speaking with @rconde01 over Slack about this, here is the test I created which is in fact failing. This rather than a bug in the code itself is something we haven't implemented yet

```py
def test_create_build_missing_negation():
    tc = TestClient(light=True)
    tc.save({"dep/conanfile.py": GenConanfile("dep", "1.0"),
             "lib/conanfile.py": GenConanfile("lib", "1.0").with_requires("dep/1.0"),
             "pkg/conanfile.py": GenConanfile("pkg", "1.0").with_requires("lib/1.0")})

    tc.run("export dep")
    tc.run("export lib")
    tc.run("create pkg --build=missing:~dep/*", assert_error=True)

    tc.assert_listed_binary({"pkg/1.0": ("a72376edfbbdaf97c8608b5fda53cadebac46a20", "Build"),
                             "lib/1.0": ("abfcc78fa8242cabcd1e3d92896aa24808c789a3", "Build"),
                             "dep/1.0": ("da39a3ee5e6b4b0d3255bfef95601890afd80709", "Missing")})

    tc.run("create pkg --build=missing:~dep/* --build=missing:~lib/*",
           assert_error=True)

    tc.assert_listed_binary({"pkg/1.0": ("a72376edfbbdaf97c8608b5fda53cadebac46a20", "Build"),
                             "lib/1.0": ("abfcc78fa8242cabcd1e3d92896aa24808c789a3", "Missing"),
                             "dep/1.0": ("da39a3ee5e6b4b0d3255bfef95601890afd80709", "Missing")})
```

**Comment by memsharded** [comment]
I am proposing https://github.com/conan-io/conan/pull/16327 as a possible implementation of this behavior.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16327 merged, will be in next 2.4
